docs/architecture/adr-035-rosetta-api-support.md

```diff
@@ -0,0 +1,225 @@
+# ADR 035: Rosetta API Support
+
+## Context
+
+[Rosetta API](https://www.rosetta-api.org/) is an open-source specification and set of tools developed by Coinbase to standardise blockchain interactions.
+
+Through the use of a standard API for integrating blockchain applications it will
+
+* Be easier for a user to interact with a given blockchain
+* Allow exchanges to integrate new blockchains quickly and easily
+* Enable application developers to build cross-blockchain applications such as block explorers, wallets and dApps at considerably lower cost and effort.
+
+## Decision
+
+It is clear that adding Rosetta API support to the Cosmos SDK will bring value to all the developers and Cosmos SDK based chains in the ecosystem. How it is implemented is key.
+
+The driving principles of the proposed design are:
+
+1. **Extensibility:** it must be as riskless and painless as possible for application developers to set-up network configurations to expose Rosetta API-compliant services.
+2. **Long term support:** This proposal aims to provide support for all the supported Cosmos SDK release series.
+3. **Cost-efficiency:** Backporting features from `master` to the various stable branches of Cosmos SDK is a cost that needs to be reduced.
```

Comment:
What does this have to do with Rosetta? Do you mean Rosetta features?

Reply:
Yep!

Reply:
Can you state that?

Reply:
Hold on - these are just principles. What exactly do you reckon should be changed?

Reply:
`Backporting Rosetta features...`

Reply:
Thanks, it's much clearer now and yes, I believe we all are on the same page here.

> (I have no idea what features)

Good point - it's not about *features*, Rosetta API specifications are quite new and as such they tend to change often. We expect various changes to the specifications to come through in the following months. Backports of such features would need to be provided and that's what we are trying to avoid here.

If you are OK with that, I'd move on amending the text.